# Analyzing CIA Factbook Data Using SQL
In this project, we'll work with data from the [CIA World Factbook](https://www.cia.gov/library/publications/the-world-factbook/), a compendium of statistics about all of the countries on Earth. The Factbook contains demographic information like:
* name - The name of the country.
* area- The country's total area (both land and water).
* area_land - The country's land area in square kilometers.
* area_water - The country's waterarea in square kilometers.
* population - The country's population.
* population_growth- The country's population growth as a percentage.
* birth_rate - The country's birth rate, or the number of births a year per 1,000 people.
* death_rate - The country's death rate, or the number of death a year per 1,000 people.

In [9]:
# Import SQL?
%%capture
%load_ext sql
%sql sqlite:///factbook.db

In [10]:
# Use %%sql to use sql
# Exploratory steps
%%sql
SELECT *
    FROM facts
 LIMIT 5;

 * sqlite:///factbook.db
Done.


[(1, 'af', 'Afghanistan', 652230, 652230, 0, 32564342, 2.32, 38.57, 13.89, 1.51),
 (2, 'al', 'Albania', 28748, 27398, 1350, 3029278, 0.3, 12.92, 6.58, 3.3),
 (3, 'ag', 'Algeria', 2381741, 2381741, 0, 39542166, 1.84, 23.67, 4.31, 0.92),
 (4, 'an', 'Andorra', 468, 468, 0, 85580, 0.12, 8.13, 6.96, 0.0),
 (5, 'ao', 'Angola', 1246700, 1246700, 0, 19625353, 2.78, 38.78, 11.49, 0.46)]

In [11]:
%%sql
SELECT MIN(population), 
       MAX(population), 
       MIN(population_growth),
       MAX(population_growth)
    FROM facts;

 * sqlite:///factbook.db
Done.


[(0, 7256490011, 0.0, 4.02)]

We can see that the there's countries with 0 and 7256490011 people which is impossible.

In [22]:
%%sql 
SELECT name
    FROM facts
 WHERE population = (SELECT MIN(population)
                        FROM facts
                    );

 * sqlite:///factbook.db
Done.


[('Antarctica',)]

In [18]:
%%sql
SELECT name
    FROM facts
 WHERE population = (SELECT MAX(population)
                         FROM facts
                    );

 * sqlite:///factbook.db
Done.


[('World',)]

Antarctica and the World are both extremes of the population and aren't really coinutries anyways so we just exclude them.

In [19]:
# Reselect the columns stats without the extremes
%%sql
SELECT MIN(population), 
       MAX(population), 
       MIN(population_growth),
       MAX(population_growth)
    FROM facts
 WHERE population > 0
 AND population < (SELECT MAX(population)
                         FROM facts
                );

 * sqlite:///factbook.db
Done.


[(48, 1367485388, 0.0, 4.02)]

In [20]:
%%sql 
SELECT AVG(population), AVG(area)
    FROM facts;

 * sqlite:///factbook.db
Done.


[(62094928.32231405, 555093.546184739)]

In [21]:
%%sql 
SELECT *
    FROM facts
 WHERE population > (SELECT AVG(population)
                        FROM facts
                    )
 AND area < (SELECT AVG(area)
                FROM facts
            );

 * sqlite:///factbook.db
Done.


[(14, 'bg', 'Bangladesh', 148460, 130170, 18290, 168957745, 1.6, 21.14, 5.61, 0.46),
 (65, 'gm', 'Germany', 357022, 348672, 8350, 80854408, 0.17, 8.47, 11.42, 1.24),
 (85, 'ja', 'Japan', 377915, 364485, 13430, 126919659, 0.16, 7.93, 9.51, 0.0),
 (138, 'rp', 'Philippines', 300000, 298170, 1830, 100998376, 1.61, 24.27, 6.11, 2.09),
 (173, 'th', 'Thailand', 513120, 510890, 2230, 67976405, 0.34, 11.19, 7.8, 0.0),
 (185, 'uk', 'United Kingdom', 243610, 241930, 1680, 64088222, 0.54, 12.17, 9.35, 2.54),
 (192, 'vm', 'Vietnam', 331210, 310070, 21140, 94348835, 0.97, 15.96, 5.93, 0.3)]